recipe/meta.yaml

```diff
@@ -109,53 +121,57 @@ outputs:
         {% endfor %}
 
         # pkg-config (no metadata for vendored libs)
-        # should work on windows in principle, but our openssl builds don't have a .pc file
+        # a number of dependencies do not have pkgconfig files on windows. e.g. re2
         {% for each_lib in core_libs %}
-        - pkg-config --print-errors --exact-version "{{ core_version }}" {{ each_lib }}  # [unix]
+        - pkg-config --print-errors --exact-version "{{ core_version }}" {{ each_lib }} # [not win]
         {% endfor %}
         {% for each_lib in core_cpp_libs %}
-        - pkg-config --print-errors --exact-version "{{ version }}" {{ each_lib }}       # [unix]
+        - pkg-config --print-errors --exact-version "{{ version }}" {{ each_lib }} # [not win]
         {% endfor %}
 
+        # CMake test: compile upstream example
         - ./test_grpc.sh   # [unix]
         - ./test_grpc.bat  # [win]
 
-  - name: {{ name }}io
-    script: build-grpcio.sh   # [not win]
+        # test availability of symbol needed by arrow (esp. on windows)
+        - cd cmake_test
+        - cmake -GNinja -DCMAKE_CXX_STANDARD=17 $CMAKE_ARGS .   # [unix]
+        - cmake -GNinja -DCMAKE_CXX_STANDARD=17 %CMAKE_ARGS% .  # [win]
+        - cmake --build .
+
+  - name: grpcio
+    script: build-grpcio.sh  # [not win]
     script: build-grpcio.bat  # [win]
     build:
       missing_dso_whitelist:  # [s390x]
-        - '$RPATH/ld64.so.1'  # [s390x]
+        - $RPATH/ld64.so.1    # [s390x]
     requirements:
       build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - cython <3                              # [build_platform != target_platform]
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - m2-patch  # [win]
-        - patch  # [not win]
       host:
-        - {{ pin_subpackage('grpc-cpp', exact=True) }}
         - python
-        - cython 0.29
         - pip
-        - pthread-stubs 0.3  # [linux]
         - setuptools
-        - wheel >=0.29
-        - abseil-cpp 20230802.0
-        - c-ares 1.19.1
-        - libprotobuf {{ libprotobuf }}
```

Comment:
Is `protobuf` required for building `grpcio`? Or it's only for examples and/or testing?
See https://github.com/grpc/grpc/blob/v1.62.2/requirements.txt#L4

Reply:
Not from what I see.